### PR TITLE
Use latest rubocop and update rubies on travis

### PR DESCRIPTION
**Pull request by mjobin-mdsol**
also deprecate ruby 2.3, 2.4 and add 3.0

### CODE REVIEW

**Review by ykitamura-mdsol — APPROVED** [review]
Looks good to me 👍 

We will drop ruby versions in gemspec when we have a version bump next time, right?
https://github.com/mdsol/dice_bag/blob/246df9dbf46d8d2ef454f16835ac6b8aacf612ea/dice_bag.gemspec#L23

File: .travis.yml [hunk 1]
```diff
@@ -3,20 +3,18 @@ dist: bionic
...
 
 before_install:
   - gem update --system -N > /dev/null && echo "Rubygems version $(gem --version)"
-  - if [ $TRAVIS_RUBY_VERSION == 2.7 ]; then gem install rubocop:1.3.1; fi
 
 script:
   - bundle exec rspec --format progress
   - bundle exec cucumber --format progress
-  - if [ $TRAVIS_RUBY_VERSION == 2.7 ]; then rubocop; fi
+  - bundle exec rubocop
```
Comment (jcarres-mdsol) [inline .travis.yml:17]: I guess there was a reason not to run rubocop on every job ?

Comment (mathieujobin) [reply .travis.yml:17]: mostly because it was broken with ruby 2.3
no major issue running with all version has long as it passes :heavy_check_mark:

**Comment by mjobin-mdsol** [comment]
@ykitamura-mdsol @jcarres-mdsol I can proceed with gem version bump, and ruby minimum version bump as well.

**Review by jcarres-mdsol — APPROVED** [review]
Looks good to me

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# 1.6.0
+* Ruby 2.3 and 2.4 support dropped.
```
Comment (jfeltesse-mdsol) [inline CHANGELOG.md]: [nits] can you use the active tense for consistency? e.g. "Drop support..."

Comment (mathieujobin) [reply CHANGELOG.md]: sure @jfeltesse-mdsol I will update and merge

**Review by mathieujobin — APPROVED** [review]

**Pull request merged**